internals-js/src/__tests__/extractSubgraphsFromSupergraph.test.ts

```diff
@@ -0,0 +1,110 @@
+import { buildSupergraphSchema, extractSubgraphsFromSupergraph } from "..";
+
+test('handles types having no fields in a subgraph correctly', () => {
```

Comment:
could we add another test which exercises the path where we remove fields in a subgraph but do not remove the whole type? mainly for documentation of the behavior—i expect this is exercised in other tests.